conn.go

```diff
@@ -257,12 +252,15 @@ func (c *Conn) Receive() ([]Message, error) {
 
 // receive is the internal implementation of Conn.Receive, which can be called
 // recursively to handle multi-part messages.
-func (c *Conn) receive() ([]Message, error) {
+func (c *Conn) receive() ([]Message, *OpError) {
```

Comment:
Returning `*OpError` concretely seems dangerous in view of the possible "nil but not nil interface" mistake, unless I'm missing something about how this code works.